### PR TITLE
add Matio build

### PR DESCRIPTION
**Pull request by cmhamel**
This is a C library for opening .mat files from Matlab without the dependencies.

### CODE REVIEW

**Comment by cmhamel** [comment]
I can't get mac to work with Hdf5_jll for some reason. I don't think it's just this library since I've had issues trying to add mac support to Exodus_jll. 

Is there something special I need to do to add mac/windows support in terms of the library paths?

File: M/Matio/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	    -DMATIO_DEFAULT_FILE_VERSION=7.3 \
+        -DMATIO_MAT73:BOOL=ON \
+        -DCMAKE_INSTALL_PREFIX=${prefix} \
+        -DCMAKE_INSTALL_LIBDIR:PATH=lib \
+        -DMATIO_WITH_HDF5:BOOL=ON \
+        -DMATIO_WITH_ZLIB:BOOL=ON \
+        -DHDF5_ROOT:PATH=${prefix} \
+        -DHDF5_DIR:PATH=${prefix} \
+	    -DZLIB_INCLUDE_DIR:PATH=${prefix}/include \
+        -DZLIB_LIBRARY:FILEPATH=${prefix}/lib/libz.${LD_EXT}
```
Comment (giordano) [inline M/Matio/build_tarballs.jl:39]: I'd start by reading the documentation: https://docs.binarybuilder.org/stable/build_tips/#CMake-builds.  In particular, `-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}` is ***not*** optional

File: M/Matio/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+cd matio
+
+# trying with cmake
+
+OS=$(uname -s)
+if [ "$OS" = "Darwin" ] ; then
+    LD_EXT="dylib"
+else
+    LD_EXT="so"
+fi
```
Comment (giordano) [inline M/Matio/build_tarballs.jl:25]: https://docs.binarybuilder.org/stable/build_tips/#Automatic-environment-variables
```suggestion
```

File: M/Matio/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        -DMATIO_MAT73:BOOL=ON \
+        -DCMAKE_INSTALL_PREFIX=${prefix} \
+        -DCMAKE_INSTALL_LIBDIR:PATH=lib \
+        -DMATIO_WITH_HDF5:BOOL=ON \
+        -DMATIO_WITH_ZLIB:BOOL=ON \
+        -DHDF5_ROOT:PATH=${prefix} \
+        -DHDF5_DIR:PATH=${prefix} \
+        -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}
+	    # -DZLIB_INCLUDE_DIR:PATH=${prefix}/include \
+        # -DZLIB_LIBRARY:FILEPATH=${prefix}/lib/libz.${LD_EXT}
```
Comment (giordano) [inline M/Matio/build_tarballs.jl:34]:
```suggestion
cmake . \
        -DBUILD_SHARED_LIBS:BOOL=ON \
        -DMATIO_SHARED:BOOL=ON \
	-DMATIO_DEFAULT_FILE_VERSION=7.3 \
        -DMATIO_MAT73:BOOL=ON \
        -DCMAKE_INSTALL_PREFIX=${prefix} \
        -DMATIO_WITH_HDF5:BOOL=ON \
        -DMATIO_WITH_ZLIB:BOOL=ON \
        -DHDF5_ROOT:PATH=${prefix} \
        -DHDF5_DIR:PATH=${prefix} \
        -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}
```

File: M/Matio/build_tarballs.jl (hunk 1 above)
Comment (cmhamel) [reply M/Matio/build_tarballs.jl]: Thanks @giordano that was very helpful.

File: M/Matio/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        -DCMAKE_INSTALL_PREFIX=${prefix} \
+        -DMATIO_WITH_HDF5:BOOL=ON \
+        -DMATIO_WITH_ZLIB:BOOL=ON \
+        -DHDF5_ROOT:PATH=${prefix} \
+        -DHDF5_DIR:PATH=${prefix} \
+        -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}
+
+cmake --build .
+cmake --install .
+
```
Comment (giordano) [inline M/Matio/build_tarballs.jl:35]: Please build inside a subdirectory.  CMake makes it incredibly inconvenient to build in the same directory where the source code is
```suggestion
cd matio
mkdir build && cd build
cmake .. \
        -DBUILD_SHARED_LIBS:BOOL=ON \
        -DMATIO_SHARED:BOOL=ON \
	-DMATIO_DEFAULT_FILE_VERSION=7.3 \
        -DMATIO_MAT73:BOOL=ON \
        -DCMAKE_INSTALL_PREFIX=${prefix} \
        -DMATIO_WITH_HDF5:BOOL=ON \
        -DMATIO_WITH_ZLIB:BOOL=ON \
        -DHDF5_ROOT:PATH=${prefix} \
        -DHDF5_DIR:PATH=${prefix} \
        -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}
make -j${nproc}
make install
```

**Pull request merged**